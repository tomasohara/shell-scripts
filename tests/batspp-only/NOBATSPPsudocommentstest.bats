#!/usr/bin/env bats


# Make executables ./tests/../ visible to PATH
PATH="/home/aveey/tom-project/shell-scripts/tests/batspp-only/../:$PATH"

# Enable aliases
shopt -s expand_aliases



@test "test-1" {
	testfolder="/tmp/batspp-85745/test-1"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'TEMP_PATH="./temp/kernelinfo.txt"\n' "=========="
	test-1-actual 
	echo "=========" $'' "========="
	test-1-expected 
	echo "============================"
	# ???: 'TEMP_PATH="./temp/kernelinfo.txt"\n'=$(test-1-actual)
	# ???: ''=$(test-1-expected)
	[ "$(test-1-actual)" == "$(test-1-expected)" ]
}

function test-1-actual () {
	# no-op in case content just a comment
	true

	TEMP_PATH="./temp/kernelinfo.txt"

}

function test-1-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED

END_EXPECTED
}


@test "test-3" {
	testfolder="/tmp/batspp-85745/test-3"
	mkdir --parents "$testfolder"
	command cp ./*.* "$testfolder"
	builtin cd "$testfolder" || echo Warning: Unable to "cd $testfolder"

	echo "==========" $'sudo -S apt search caffeine < ./temp/pass.txt\n' "=========="
	test-3-actual 
	echo "=========" $'Sorting... 0%rd for aveey: Sorting... 0%Sorting... 0%Sorting... Done\nFull Text Search... 50%Full Text Search... 50%Full Text Search... Done\n\x1b[32mcaffeine\x1b[0m/jammy,jammy,now 2.9.8-2 all [residual-config]\n  prevent the desktop becoming idle in full-screen mode' "========="
	test-3-expected 
	echo "============================"
	# ???: 'sudo -S apt search caffeine < ./temp/pass.txt\n'=$(test-3-actual)
	# ???: 'Sorting... 0%rd for aveey: Sorting... 0%Sorting... 0%Sorting... Done\nFull Text Search... 50%Full Text Search... 50%Full Text Search... Done\n\x1b[32mcaffeine\x1b[0m/jammy,jammy,now 2.9.8-2 all [residual-config]\n  prevent the desktop becoming idle in full-screen mode'=$(test-3-expected)
	[ "$(test-3-actual)" == "$(test-3-expected)" ]
}

function test-3-actual () {
	# no-op in case content just a comment
	true

	sudo -S apt search caffeine < ./temp/pass.txt

}

function test-3-expected () {
	# no-op in case content just a comment
	true

	cat <<END_EXPECTED
Sorting... 0%rd for aveey: Sorting... 0%Sorting... 0%Sorting... Done
Full Text Search... 50%Full Text Search... 50%Full Text Search... Done
[32mcaffeine[0m/jammy,jammy,now 2.9.8-2 all [residual-config]
  prevent the desktop becoming idle in full-screen mode
END_EXPECTED
}
