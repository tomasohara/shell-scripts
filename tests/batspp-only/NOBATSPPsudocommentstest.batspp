# uname -r | sudo tee $TEMP

$ TEMP_PATH="./temp/kernelinfo.txt"

$ sudo -S apt search caffeine < ./temp/pass.txt
Sorting... 0%rd for aveey: Sorting... 0%Sorting... 0%Sorting... Done
Full Text Search... 50%Full Text Search... 50%Full Text Search... Done
[32mcaffeine[0m/jammy,jammy,now 2.9.8-2 all [residual-config]
  prevent the desktop becoming idle in full-screen mode

[32mfonts-aenigma[0m/jammy,jammy 0.0.20080511+dfsg-4 all
  465 free TrueType fonts by Brian Kent

[32mlibcaffeine-java[0m/jammy,jammy 2.6.2-1.1 all
  High performance caching library
