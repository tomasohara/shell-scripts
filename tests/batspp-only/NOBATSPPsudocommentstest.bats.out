1..2
ok 1 test-1
not ok 2 test-3
# (from function `test-3-actual' in file batspp-only/NOBATSPPsudocommentstest.bats, line 66,
#  in test file batspp-only/NOBATSPPsudocommentstest.bats, line 53)
#   `test-3-actual ' failed
# ========== sudo -S apt search caffeine < ./temp/pass.txt
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-only/NOBATSPPsudocommentstest.bats: line 66: ./temp/pass.txt: No such file or directory
